### ISSUE TITLE
Add support for statically linking Jansi JNI library

### ISSUE THREAD

**Issue by rsenden** [fusesource/jansi#246]
We have a picocli-based application for which we build native images for Windows, Linux & MacOS using GraalVM. For various reasons, we build a statically linked executable for Linux, and dynamically linked executables for Windows/MacOS.

We would like to add Jansi (and JLine) to our application. Jansi 2.4.0 works just fine for our dynamically linked executables, however it fails to load the native jansi library in the statically linked executable because `musl` apparently doesn't support dynamic library loading (see https://github.com/oracle/graal/issues/4028#issue-1055983935):

```
open("/tmp", O_RDONLY|O_LARGEFILE|O_CLOEXEC|O_DIRECTORY) = 4
open("/tmp/jansi-2.4.0-83026643492bd4eb-libjansi.so.lck", O_WRONLY|O_CREAT|O_TRUNC|O_LARGEFILE, 0666) = 4
open("/tmp/jansi-2.4.0-83026643492bd4eb-libjansi.so", O_WRONLY|O_CREAT|O_TRUNC|O_LARGEFILE, 0666) = 4
open("/tmp/jansi-2.4.0-83026643492bd4eb-libjansi.so", O_RDONLY|O_LARGEFILE) = 4
Failed to load native library:jansi-2.4.0-83026643492bd4eb-libjansi.so. osinfo: Linux/x86_64
java.lang.UnsatisfiedLinkError: Can't load library: /tmp/jansi-2.4.0-83026643492bd4eb-libjansi.so
```

Apart from this issue, it's also just more logical to statically link the Jansi library when building native images, to avoid dynamic extraction of the library to a temporary directory and then loading it.

Based on what I've read so far, I think the following would be needed:
- jansi.c: Export either `JNI_OnLoad` (for dynamic linking) or `JNI_OnLoad_jansi` (for static linking)
- Update `Makefile` to add support for building `libJansi.a`/`jansi.lib` files
- Optional: publish pre-built `.a`/`.lib` files, to avoid users having to build these themselves
- Update JansiLoader to (also) try `System.loadLibrary("jansi")` (either always try, or use a system property to decide between this and the original `System.load(path)`-based approach)
- Document how to statically link Jansi:
    - Add `org.graalvm.nativeimage:svm` dependency
    - Add the required GraalVM `@AutomaticFeature` (alternatively, publish a new `jansi-<version>-static-feature.jar` artifact that readily includes this feature, such that users can simply add this as a dependency)
    - Add `-H:CLibraryPath=/path/to/dir/containing/libJansi.a` option to `native-image` invocation
    - Add a resource exclude to prevent dynamic Jansi libraries being added to the native image, basically making GraalVM ignore: https://github.com/fusesource/jansi/blob/00e15cb1fab9117790580a412f7d54f2c08d2f11/src/main/resources/META-INF/native-image/jansi/resource-config.json#L5

For reference, some links related to statically linking JNI libraries:
- https://www.blog.akhil.cc/static-jni
- https://github.com/oracle/graal/issues/3359#issuecomment-822070278
- https://github.com/oracle/graal/issues/3359#issuecomment-1031482202

**Comment by xtaixe** [comment]
+1. We are also facing this issue.

**Comment by gnodet** [comment]
A PR would be welcomed @rsenden

**Comment by Glavo** [comment]
Let me give it a try, I'm doing some work on this.

**Comment by rsenden** [comment]
Thanks for looking into this!

**Comment by gnodet** [comment]
@Glavo you're still planning to work on this feature ? If that's the case, I can wait a bit before starting a 2.5.0 release.

**Comment by Glavo** [comment]
> @Glavo you're still planning to work on this feature ? If that's the case, I can wait a bit before starting a 2.5.0 release.

I don't think it can be done in a short time. If I don't find a workaround, I need to wait for GraalVM to solve [#7507](https://github.com/oracle/graal/issues/7507), or give up this method and explore other methods.

**Comment by Glavo** [comment]
hmm, I found another easier way. I'll give it a quick try and if it works I'll report the results here.

**Comment by Glavo** [comment]
I think the #269 way is more promising, please wait for me.
